datafusion/execution/src/cache/cache_unit.rs

```diff
@@ -111,64 +113,224 @@ impl CacheAccessor<Path, Arc<Statistics>> for DefaultFileStatisticsCache {
 ///
 /// Collected files metadata for listing files.
 ///
-/// Cache is not invalided until user calls [`Self::remove`] or [`Self::clear`].
+/// # Internal details
+///
+/// The `capacity` parameter controls the maximum number of entries in the cache, using a Least
+/// Recently Used eviction algorithm. When adding a new entry, if the total number of entries in
+/// the cache exceeds `capacity`, the least recently used entries are evicted until the total
+/// entries are lower than the `capacity`.
 ///
 /// [`ListFilesCache`]: crate::cache::cache_manager::ListFilesCache
 #[derive(Default)]
 pub struct DefaultListFilesCache {
-    statistics: DashMap<Path, Arc<Vec<ObjectMeta>>>,
+    state: Mutex<DefaultListFilesCacheState>,
+}
+
+impl DefaultListFilesCache {
+    pub fn new(capacity: usize, ttl: Duration) -> Self {
+        Self {
+            state: Mutex::new(DefaultListFilesCacheState::new(capacity, ttl)),
+        }
+    }
+
+    pub fn cache_limit(&self) -> usize {
+        self.state.lock().unwrap().capacity
+    }
+
+    pub fn cache_ttl(&self) -> Duration {
+        self.state.lock().unwrap().ttl
+    }
+}
+
+pub(super) const DEFAULT_LIST_FILES_CACHE_LIMIT: usize = 128 * 1024; // ~130k objects
+pub(super) const DEFAULT_LIST_FILES_CACHE_TTL: Duration = Duration::new(600, 0); // 10min
+
+pub struct DefaultListFilesCacheState {
+    lru_queue: LruQueue<Path, (Arc<Vec<ObjectMeta>>, Instant)>,
+    capacity: usize, // TODO: do "bytes" matter here, or should we stick with "entries"?
```

Comment:
I didn't feel like limiting this cache by "bytes" really made sense because the data stored in the cache is generally very uniform in size, perhaps aside from the path. I felt that it was probably small enough that simply limiting it by the number of entries should suffice, and "entries" seems like it would be easier for users to configure.

Reply:
I think that bytes makes more sense, mostly because that is the resource that is actually used. count of items is a proxy for resource usage